File: exo5/exo5.rb
mot = "beweb"  #déclaration des variables
mot.downcase!
choix=2

puts case choix  #switch pour les differentes options
when 1    #option palindrome
  deb_mot=0
  fin_mot=mot.length-1
  palindrome=1
  	while(deb_mot< fin_mot)
        if mot[deb_mot]!=mot[fin_mot]
          palindrome=0
          break
        end
        deb_mot = deb_mot +1
        fin_mot = fin_mot -1
    end
  if(palindrome==1)
    print("C'est un palindrome")
  else
    print("Ce n'est pas un palindrome")
  end
when 2      #compte le nombre de caractère alphabétique
  nb_lettre = 0
  i=0
    while i < mot.length
      if(/[a-z]/ !~ mot[i])==false  #condition qui est vrai seulemement si mot[i] est une lettre
        nb_lettre=nb_lettre +1
      end
      i = i+1
    end
  print(nb_lettre)
when 3    #remplacement d'une lettre par une autre
  remp = "z"
  val = "w"
  i=0
  while i < mot.length
    if mot[i]==val     # si on tombe sur la lettre a remplacer, alors on rentre dans le si
     mot[i]=remp       # on remplace la lettre par celle voulu
    end
    i = i +1
  end
  print(mot)
when 4  #séparer le mot
  print(mot.chars.each_slice(mot.length/2).map(&:join)) #permet de split un texte en 2 ou plus
  # print(mot.split("w").last)



end

File: exo5/exo5.java
public class exo5 {
    public static void main(String[] args) {
        String mot = "beweb";
        mot = mot.toLowerCase();
        int choix = 2;
        Boolean palindrome = true;
        char[] tab_mot = mot.toCharArray();

        if (choix == 1) {
            int deb_mot = 0;
            int fin_mot = mot.length()-1;
            while ( deb_mot < fin_mot & palindrome == true ) {
                tab_mot = mot.toCharArray();

                if (tab_mot[deb_mot] != tab_mot[fin_mot]) {
                    palindrome = false;
                }
                deb_mot ++;
                fin_mot --;
            }
            if (palindrome) {
                System.out.print("C'est un palindrome\n");
            } else { System.out.print("Ce n'est pas un palindrome\n"); }
        }

        if (choix == 2) {
            int nb_alpha=0;
            for (int i=0; i < tab_mot.length ; i++) {
                  String test = Character.toString(tab_mot[i]);
                if ( test.matches("^[a-z]+") ) {
                    nb_alpha++;
                }

            }
            System.out.print(nb_alpha + "\n");

        }
        if (choix == 3) {
    String remp = "z";
    String val = "w";
    for (int i=0; i < tab_mot.length; i++) {
        String lettre = Character.toString(tab_mot[i]);
        if ( lettre == val ) {
            System.out.print("test");
            String lettre2 = Character.toString(remp);
            tab_mot[i] = lettre2;
        }

    }

}
    }
}
